static/app/utils/discover/genericDiscoverQuery.tsx

```diff
@@ -199,6 +199,9 @@ class _GenericDiscoverQuery<T, P> extends React.Component<Props<T, P>, State<T>>
 
     beforeFetch?.(api);
 
+    // clear any inflight requests since they are now stale
+    api.clear();
```

Comment:
Does this have the potential to cancel other requests unintentionally? 🤔

Reply:
The cancellation only occurs for the `Client` instance of `api`. Not other `Client` instances.

Reply:
basically it depends on how you pass in `api`, I think perhaps we should make these instances unique now with `useAPI()`